pkg/store/consul/transaction/transaction.go

```diff
@@ -123,3 +124,23 @@ func txnErrorsToString(errors api.TxnErrors) string {
 
 	return str
 }
+
+func getTxnFromContext(ctx context.Context) (*Tx, error) {
+	select {
+	case <-ctx.Done():
+		return nil, ErrAlreadyCommitted
+	default:
+	}
+
+	txnValue := ctx.Value(contextKey)
+	if txnValue == nil {
+		return nil, util.Errorf("no transaction was opened on the passed Context")
```

Comment:
removed because we're never going to check for that type of error, it indicates that the code is just written wrong!